### ISSUE TITLE
Fatal packaging QA complains and build publishing rejection in openSUSE

### ISSUE THREAD

**Issue by v-fox** [lpereira/hardinfo#298]
So, I've tried [making OBS package](https://build.opensuse.org/package/show/home:X0F:HSF/hardinfo) for latest snapshot of hardinfo as replacement for [ancient openSUSE build](https://build.opensuse.org/package/show/utilities/hardinfo) but after successfully building it, OBS refused to publish it by stating these segfault-threatening security/stability concerns:
```
[   67s] I: Program is using implicit definitions of functions getting
[   67s]    pointers or implemented by macros. These functions need to use their
[   67s]    correct prototypes to allow correct argument passing on e.g. x86_64 .
[   67s]      - Implicit memory/string functions need #include <string.h>.
[   67s]      - Implicit *printf functions need #include <stdio.h>.
[   67s]      - Implicit *printf functions need #include <stdio.h>.
[   67s]      - Implicit *read* functions need #include <unistd.h>.
[   67s]      - Implicit *recv* functions need #include <sys/socket.h>.
[   67s] W: hardinfo implicit-pointer-decl /home/abuild/rpmbuild/BUILD/hardinfo-20181022T194523/hardinfo/dt_util.c:540
[   67s] W: hardinfo implicit-pointer-decl /home/abuild/rpmbuild/BUILD/hardinfo-20181022T194523/hardinfo/vendor.c:119
[   67s] W: hardinfo implicit-pointer-decl /home/abuild/rpmbuild/BUILD/hardinfo-20181022T194523/modules/benchmark/bench_results.c:127, 208
[   67s] W: hardinfo implicit-pointer-decl /home/abuild/rpmbuild/BUILD/hardinfo-20181022T194523/modules/computer/os.c:396
[   67s] W: hardinfo implicit-pointer-decl /home/abuild/rpmbuild/BUILD/hardinfo-20181022T194523/modules/network.c:152
[   67s] 
[   67s] I: Program is using uninitialized variables.
[   67s]    Note the difference between "is used" and "may be used"
[   67s] W: hardinfo uninitialized-variable /home/abuild/rpmbuild/BUILD/hardinfo-20181022T194523/modules/benchmark/zlib.c:28
[   67s] 
[   67s] I: Program returns random data in a function
[   67s] E: hardinfo no-return-in-nonvoid-function /home/abuild/rpmbuild/BUILD/hardinfo-20181022T194523/hardinfo/gpu_util.c:37, 405
[   67s] E: hardinfo no-return-in-nonvoid-function /home/abuild/rpmbuild/BUILD/hardinfo-20181022T194523/hardinfo/pci_util.c:202, 160
[   67s] E: hardinfo no-return-in-nonvoid-function /home/abuild/rpmbuild/BUILD/hardinfo-20181022T194523/modules/benchmark/bench_results.c:137
[   67s] E: hardinfo no-return-in-nonvoid-function /home/abuild/rpmbuild/BUILD/hardinfo-20181022T194523/modules/devices/spd-decode.c:982
[   67s] 
[   67s] I: Program returns random data in a function
[   67s] E: hardinfo no-return-in-nonvoid-function /home/abuild/rpmbuild/BUILD/hardinfo-20181022T194523/hardinfo/gpu_util.c:37, 405
[   67s] E: hardinfo no-return-in-nonvoid-function /home/abuild/rpmbuild/BUILD/hardinfo-20181022T194523/hardinfo/pci_util.c:202, 160
[   67s] E: hardinfo no-return-in-nonvoid-function /home/abuild/rpmbuild/BUILD/hardinfo-20181022T194523/modules/benchmark/bench_results.c:137
[   67s] E: hardinfo no-return-in-nonvoid-function /home/abuild/rpmbuild/BUILD/hardinfo-20181022T194523/modules/devices/spd-decode.c:982
```

**Comment by bp0** [comment]
Thanks. I'll look into this later today.

**Comment by bp0** [comment]
Fixed by #299.
